portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```
@@ -64,14 +60,17 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     }
 
     /** Parses user-requested max number of comments, returned num is valid and positive */
-    private int getBoundedMaxComments(HttpServletRequest request) {
+    private int getBoundedMaxComments(HttpServletRequest request, int numComments) {
         int requestedComments = 5;
-        int defaultComments = 5;
         String inputMaxComments = request.getParameter("max-comments");
-        if (inputMaxComments != null) {
+        try {
             requestedComments = Integer.parseInt(inputMaxComments);
-        } 
-        return ((requestedComments >= 0) ? requestedComments : defaultComments);
+        } catch (Exception e) { /* ignore */ }
```

It's often a bad idea to swallow exceptions by doing nothing with them, the minimal to do would be to log it or display a message that something went wrong so you or the user knows about it. It might be a bit of extra work to add a message for that, so in the meantime you can add a comment that it retains the default and also a TODO note so you'll remember for later.

It's also usually a bad idea to do a catch-all on Exception without specifying specific exceptions (what if the exception is from a case we didn't expect, and we ignored it when we shouldn't?)
parseInt throws NumberFormatException when the input is null or invalid, so you'll want to catch just that instead.

---

btw not sure if you've seen the to-do format yet, but it basically goes like
`// TODO(cmargaret): some message`

---

Every time the page is refreshed, the request is null for the comments number since I'm calling a GET request onload. I want the comments to reload when the page is reloaded but that means it also checks for the number of comments. Is there a way I can separate out that logic / detect when it's a reload vs a change in requested comment number?

---

Embedding the value in a URL param may be one way as that value would not go away upon refresh (not too familar with the best practice there myself), you'd just get the value from the URL on every page load. Updating the number of comments would just reload the page with another value in the URL. 

Other things I could think of would be to use cookies or a cache, but those are probably way too much trouble for just this.

---

Yeah I think embedding it into the url would be most reasonable for this. I can look into implementing that

---

added the feature, now keeps the same amount of comments shown after refresh.

---

btw not sure if you've sent out the review for the latest changes yet but I can't access it.

---

Oh, I added it as a new commit on this same branch which should show up on the pull request. Commit abdf9cb is the latest one, can you access that one?

---

I can see the specific changes on that commit (only js changes) but unable to review it, I'm still not fully familiar with Git but you may have to somehow re-request a review for this pull request?

---

I just re-requested a review. Can you access it?

---

Yup! Sent out some more comments just now.